src/widget/axis_size.rs

```diff
@@ -0,0 +1,27 @@
+use super::container::Axis;
+use embedded_graphics::prelude::*;
+
+pub trait AxisSize<T> {
+    fn for_axis(&self, axis: Axis) -> T;
+}
```

Comment:
For both `Size` and `IntrinsicSize`, get the dimension relevant to a specific axis.